protos/orchestrator_service.proto

```diff
@@ -681,4 +690,16 @@ message CompleteTaskResponse {
 
 message HealthPing {
 	// No payload
+}
+
+message StreamInstanceHistoryRequest {
+    string instanceId = 1;
+    google.protobuf.StringValue executionId = 2;
+
+    // When set to true, the service may return a more optimized response suitable for workers.
+    bool forWorkItemProcessing = 3;
+}
+
+message HistoryChunk {
```

Comment:
another question: how will the client know when all messages have been received?

Reply:
The server will close the stream when it's done sending everything. In C#, the stream is closed by letting the service API method complete. From the perspective of the C# client, closing the stream causes the `await foreach` loop to exit.